frontend/src/components/public_group/AmountPicker.js

```diff
@@ -42,45 +41,15 @@ export default class AmountPicker extends React.Component {
             // MAJOR HACK to support a donation for this group.
             showCurrencyPicker={group.id == 10}/>
         )}
-        <div className='max-width-1 mx-auto' style={{position: 'absolute', left: '0', right: '0', margin: '-25px auto'}}>
-          {hasPaypal && (
-            <AsyncButton
-              color={hasPaypal ? 'green' : ''}
-              inProgress={inProgress}
-              customClass='btn -btn-big -bg-green -ttu -ff-sec -fw-bold'
-              onClick={() => onToken({
-                amount,
-                frequency,
-                currency,
-                options: {
-                  paypal: true
-                }
-              })}>
-              {i18n.getString('donateWithPayPal')}
-            </AsyncButton>
-          )}
-
-          {hasStripe && !hasPaypal && ( // paypal has priority -> to refactor after prototyping phase
-            <StripeCheckout
-              token={(token) => onToken({amount, frequency, currency, token})}
-              stripeKey={stripeKey}
-              name={group.name}
-              currency={currency}
-              amount={convertToCents(amount)}
-              description={stripeDescription}>
-                <AsyncButton
-                  color='green'
-                  inProgress={inProgress}
-                  customClass='btn -btn-big -bg-green -ttu -ff-sec -fw-bold'>
-                  {button}
-                </AsyncButton>
-            </StripeCheckout>
-          )}
-
-          <div className='PublicGroupForm-disclaimer h6 mt2' style={{minHeight: 38}}>
-            {i18n.getString('disclaimer')} {group.host.name} {stripeDescription} {i18n.getString('for')} {group.name}. {cancellationDisclaimer}
-          </div>
-        </div>
+        <DonationDistributor 
+          amount={amount}
+          method={(hasPaypal) ? 'paypal' : (hasStripe) ? 'stripe' : 'paypal'}
```

Comment:
If the collective doesn't have paypal it tests if the collective has stripe enabled otherwise it fallsback to paypal